src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs

```diff
@@ -249,7 +249,7 @@ await subscriber.Value.InitializeSubscriberAsync(this, _activeConfiguredProjectS
 
             await _commonServices.ThreadingService.SwitchToUIThread();
 
-            using (_tasksService.LoadedProject())
+            await _tasksService.LoadedProjectAsync(() =>
```

Comment:
[LoadedProjectAsync](http://index/?rightProject=Microsoft.VisualStudio.ProjectSystem&file=Utilities%5cCommonProjectSystemTools.cs&line=206) returns a JoinableTask which always resumes onto the same thread as its caller.

---
In reply to: [121514747](https://github.com/dotnet/project-system/pull/2419#discussion_r121514747) [](ancestors = 121514747)